### PR TITLE
Activities

### PR DESCRIPTION
**Pull request by pictureb00k**

### CODE REVIEW

**Comment by Scorpibear** [comment]
Need to add tests for classes which were changed

File: src/main/java/com/epam/ta/reportportal/database/CustomMongoConverters.java [hunk 1]
```diff
@@ -21,19 +21,50 @@
...
+import org.springframework.data.convert.WritingConverter;
+import org.springframework.util.ClassUtils;
 
 public class CustomMongoConverters {
 
 	private CustomMongoConverters() {
 		//statics only
 	}
 
+	class InternalAbstractConditionalEnumConverter implements ConditionalConverter {
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/database/CustomMongoConverters.java]: seems i added it by mistake. plz remove

File: src/main/java/com/epam/ta/reportportal/database/entity/item/ActivityEventType.java [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+    ActivityEventType(String value) {
+        this.value = value;
+    }
+
+    public String getValue() {
+        return value;
+    }
+
+    public static Optional<ActivityEventType> fromString(String string) {
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/database/entity/item/ActivityEventType.java]: how often do we call this method? Just as idea - we can have static map and get value from map

File: src/main/java/com/epam/ta/reportportal/database/CustomMongoConverters.java [hunk 3]
```diff
@@ -82,4 +113,44 @@ public DBObject convert(Class<?> source) {
 
 		}
 	}
+
+	public enum ActivityEventTypeToStringConverter implements Converter<ActivityEventType, String> {
```
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/database/CustomMongoConverters.java]: unit tests for converters

File: src/main/java/com/epam/ta/reportportal/database/entity/item/ActivityEventType.java (hunk 2 above)
Comment (avarabyeu) [inline src/main/java/com/epam/ta/reportportal/database/entity/item/ActivityEventType.java]: unit tests for 'fromString'

Comment (pictureb00k) [inline src/main/java/com/epam/ta/reportportal/database/entity/item/ActivityEventType.java]: It is not very often used method, so dont really need

File: src/main/java/com/epam/ta/reportportal/database/CustomMongoConverters.java (hunk 3 above)
Comment (pictureb00k) [inline src/main/java/com/epam/ta/reportportal/database/CustomMongoConverters.java]: +

File: src/main/java/com/epam/ta/reportportal/database/entity/item/ActivityEventType.java (hunk 2 above)
Comment (pictureb00k) [inline src/main/java/com/epam/ta/reportportal/database/entity/item/ActivityEventType.java]: +

File: src/main/java/com/epam/ta/reportportal/database/CustomMongoConverters.java (hunk 1 above)
Comment (pictureb00k) [inline src/main/java/com/epam/ta/reportportal/database/CustomMongoConverters.java]: +

**Comment by avarabyeu** [comment]
Could you please resolve the conflicts so i can merge it automatically?

**Pull request merged**